design-documents/graph-ql/coverage/returns.md

```diff
@@ -0,0 +1,350 @@
+## Configuration 
+
+The following settings should be accessible via `storeConfig` query:
+- Returns functionality status on the storefront: enabled/disabled
+
+Scenarios which may need these settings include:
+- Rendering of the returns section in the customer account
+
+```graphql
+{
+  storeConfig {
+    sales_magento_rma_enabled
+  }
+}
+```
+
+## Use cases
+
+### View return list with pagination in customer account
+
+```graphql
+{
+  customer {
+    returns(pageSize: 10, currentPage: 2) {
+      items {
+        id
+        creation_date
+        customer_name
+        status
+      }
+      page_info {
+        current_page
+        page_size
+        total_pages
+      }
+      total_count
+    }
+  }
+}
+```
+
+### View return list with pagination in order details
+
+```graphql
+{
+  customer {
+    orders(filter: {number: {eq: "00000008"}}) {
+      items {
+        returns(pageSize: 10, currentPage: 1) {
+          items {
+            id
+            creation_date
+            customer_name
+            status
+          }
+          page_info {
+            current_page
+            page_size
+            total_pages
+          }
+          total_count
+        }
+      }
+    }
+  }
+}
+```
+
+### View return details
+
+```graphql
+{
+  customer {
+    return(id: "0000003") {
+      id
+      order_id
+      creation_date
+      customer_email
+      customer_name
+      status
+      comments {
+        id
+        text
+        created_at
+        created_by
+      }
+      items {
+        id
+        product {
+          sku
+          name
+        }
+        custom_attributes {
```

Comment:
For this scenario, how does the UI know the formats of each different `value` field?

For example, UI developers will want to render a Yes/No field's value differently than a single line text field's value.

Reply:
Guessing a second round trip has to be made for `customAttributeMetadata`?

Reply:
For now the Idea was the developer should use `pageSpecificCustomAttributes` first to get metadata for the supported attributes.

We can also add metadata required for rendering to the `custom_attributes`.